.github/workflows/docker-verify.yaml

```diff
@@ -0,0 +1,57 @@
+name: Docker verify process
+
+on:
+  pull_request:
+
+env:
+  HADOLINT_DOCKER_IMAGE: hadolint/hadolint:v2.12.0
+
+defaults:
+  run:
+    working-directory: ./infrastructure/docker
+
+jobs:
+  gateway-lint:
```

Comment:
Should we use a [matrix](https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs) for the tests (with the dockerfile as the variable)?

Reply:
Oh, didn't know this. Yep, I think we should 👍

Reply:
Implemented on [e37f8b9](https://github.com/Qiskit-Extensions/quantum-serverless/pull/689/commits/e37f8b9f6e02b14a496db31e725ffec5fa49dbbc)